common/src/main/kotlin/io/homeassistant/companion/android/common/data/prefs/PrefsRepositoryImpl.kt

```diff
@@ -5,267 +5,289 @@ import io.homeassistant.companion.android.common.data.LocalStorage
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+@VisibleForTesting
+const val MIGRATION_PREF = "migration"
+
+@VisibleForTesting
+const val MIGRATION_VERSION = 1
+
+private const val PREF_VER = "version"
+private const val PREF_THEME = "theme"
+private const val PREF_LANG = "lang"
+private const val PREF_LOCALES = "locales"
+private const val PREF_SCREEN_ORIENTATION = "screen_orientation"
+private const val PREF_CONTROLS_AUTH_REQUIRED = "controls_auth_required"
+private const val PREF_CONTROLS_AUTH_ENTITIES = "controls_auth_entities"
+private const val PREF_CONTROLS_ENABLE_STRUCTURE = "controls_enable_structure"
+private const val CONTROLS_PANEL_SERVER = "controls_panel_server"
+private const val CONTROLS_PANEL_PATH = "controls_panel_path"
+private const val PREF_FULLSCREEN_ENABLED = "fullscreen_enabled"
+private const val PREF_KEEP_SCREEN_ON_ENABLED = "keep_screen_on_enabled"
+private const val PREF_PAGE_ZOOM_LEVEL = "page_zoom_level"
+private const val PREF_PINCH_TO_ZOOM_ENABLED = "pinch_to_zoom_enabled"
+private const val PREF_AUTOPLAY_VIDEO = "autoplay_video"
+private const val PREF_ALWAYS_SHOW_FIRST_VIEW_ON_APP_START = "always_show_first_view_on_app_start"
+private const val PREF_WEBVIEW_DEBUG_ENABLED = "webview_debug_enabled"
+private const val PREF_KEY_ALIAS = "key-alias"
+private const val PREF_CRASH_REPORTING_DISABLED = "crash_reporting"
+private const val PREF_IGNORED_SUGGESTIONS = "ignored_suggestions"
+private const val PREF_AUTO_FAVORITES = "auto_favorites"
+private const val PREF_LOCATION_HISTORY_DISABLED = "location_history"
+private const val PREF_IMPROV_PERMISSION_DISPLAYED = "improv_permission_displayed"
+private const val PREF_GESTURE_ACTION_PREFIX = "gesture_action"
+
+/**
+ * This class ensure that when we use the local storage in [PrefsRepositoryImpl] the migrations has been made
+ * before interacting with it.
+ */
+private class LocalStorageWithMigration(
+    private val localStorage: LocalStorage,
+    private val integrationStorage: LocalStorage,
+) {
+    private val migrationChecked = AtomicBoolean(false)
+    private val migrationMutex = Mutex()
+
+    private suspend fun checkMigration() {
+        migrationMutex.withLock {
+            if (!migrationChecked.get()) {
+                val currentVersion = localStorage.getInt(MIGRATION_PREF)
+                if (currentVersion == null || currentVersion < 1) {
+                    listOf(
+                        PREF_CONTROLS_AUTH_REQUIRED,
+                        PREF_CONTROLS_AUTH_ENTITIES,
+                        PREF_FULLSCREEN_ENABLED,
+                        PREF_KEEP_SCREEN_ON_ENABLED,
+                        PREF_PINCH_TO_ZOOM_ENABLED,
+                        PREF_AUTOPLAY_VIDEO,
+                        PREF_ALWAYS_SHOW_FIRST_VIEW_ON_APP_START,
+                        PREF_WEBVIEW_DEBUG_ENABLED,
+                    ).forEach { key ->
+                        integrationStorage.getString(key)?.let {
```

Comment:
You're now always using `getString` for each pref in the migration vs `getString`, `getStringSet` or `getBoolean` as needed.

Reply:
😱😱 good catch I went too fast on this. I felt too confident, I'm going to revert this change.